gamemodes/terrortown/entities/weapons/weapon_ttt_spawneditor.lua

```diff
@@ -0,0 +1,530 @@
+			local dist3d = proximitySpawn.dist3d
+			local color
+
+			scPos = proximitySpawn.scPos
+
+			if spawnType == SPAWN_TYPE_WEAPON then
+				color = colorWeapon
+			elseif spawnType == SPAWN_TYPE_AMMO then
+				color = colorAmmo
+			elseif spawnType == SPAWN_TYPE_PLAYER then
+				color = colorPlayer
+			end
+
+			if i == 1 then
+				local client = LocalPlayer()
+
+				-- make sure there is nothingin the way
+				local trace = util.TraceLine({
+					start = client:EyePos(),
+					endpos = pos,
+					filter = {client},
+					mask = MASK_SOLID
+				})
+
+				if not trace.HitWorld then
+					entspawnscript.SetFocusedSpawn(spawnType, entType, id, spawn)
+				end
+
+				renderDrawSphere(pos, sphereRadius * (trace.HitWorld and 1 or 1.1), 4 + 750 / dist3d, 4 + 750 / dist3d, ColorAlpha(color, 245))
+			else
+				renderDrawSphere(pos, sphereRadius, 4 + 750 / dist3d, 4 + 750 / dist3d, ColorAlpha(color, 100))
+			end
+		end
+
+		if not previewData.inRange then return end
+
+		local colorSphere
+
+		if previewData.inPlacement then
+			colorSphere = ColorAlpha(entspawnscript.GetColorFromSpawnType(previewData.spawnType), 245)
+
+			-- render circle on ground
+			camStart3D2D(
+				previewData.posBase + previewData.normalBase * 2,
+				previewData.normalBase:Angle() + Angle(90, 0, 0),
+				0.25
+			)
+
+				draw.Circle(0, 0, 30, colorPreview)
+
+			camEnd3D2D()
+
+			-- render line that shows shift
+			camStart3D2D(
+				previewData.posBase + previewData.normalBase * 2,
+				Angle(0, LocalPlayer():GetAngles().y - 90, 90),
+				0.25
+			)
+
+				draw.Box(-1, 0, 2, 4 * (previewData.heightShift or 0), colorPreview)
+
+			camEnd3D2D()
+		else
+			colorSphere = colorPreview
+		end
+
+		renderDrawSphere(
+			previewData.currentPos,
+			sphereRadius,
+			4 + 750 / previewData.distance3d,
+			4 + 750 / previewData.distance3d,
+			colorSphere
+		)
+	end
+
+	---
+	-- @ignore
+	function SWEP:OnRemove()
+		hook.Remove("PostDrawTranslucentRenderables", "RenderWeaponSpawnEdit")
+	end
+
+	---
+	-- @ignore
+	function SWEP:Initialize()
+		self.modes = entspawnscript.GetSpawnTypeList()
+		self.selectedMode = 1
+
+		self:AddTTT2HUDHelp("spawneditor_place", "spawneditor_remove")
+		self:AddHUDHelpLine("spawneditor_change", Key("+reload", "R"))
+		self:AddHUDHelpLine("spawneditor_ammo_edit", Key("+walk", "WALK"))
+
+		hook.Add("PostDrawTranslucentRenderables", "RenderWeaponSpawnEdit", RenderHook)
+	end
+
+	local matScreen = Material("models/weapons/v_toolgun/screen")
+	local screenSize = 256
+	local padding = 16
+	local iconSize = 64
+	local iconX = 0.5 * (screenSize - iconSize)
+	local iconY = 2 * padding
+	local textX = 0.5 * screenSize
+	local textY = iconY + iconSize + padding
+	local lineY = screenSize - 48
+	local lineW = screenSize - 2 * padding
+	local circleS = 6
+	local circleY = screenSize - 24
+
+	local RTTexture = GetRenderTarget("TTT2SpawnPlacer", screenSize, screenSize)
+
+	local colorBasic = Color(255, 255, 255, 100)
+	local colorSelect = Color(255, 255, 255, 235)
```

Comment:
Wouldnt it make sense to have all variables at the top, so you know where to look? Or is it easier to just have them above the function, thats actually using them?

Reply:
yes. However line 313 is client only so it has to stay there. Therefore I kept all of them there. But I generally agree with you. Hm

Reply:
To line 96 I mean, there are clientside only variables.